networking/src/p2p/peer.rs

```diff
@@ -438,6 +445,24 @@ pub async fn bootstrap(
     let connection_message =
         ConnectionMessage::from_bytes(received_connection_message_bytes.content())?;
 
+    // create PublicKey from received bytes from remote peer
+    let peer_public_key = PublicKey::from_bytes(connection_message.public_key())?;
+
+    let connecting_to_self = peer_public_key == info.identity.public_key;
+    if connecting_to_self {
+        debug!(log, "Detected self connection");
```

Comment:
@akoptelov 
change this debug to warn please,
if this happens a lot, then we need to fix it not to happen